src/main/java/org/broadinstitute/hellbender/tools/exome/gcbias/GCCorrector.java

```diff
@@ -0,0 +1,107 @@
+package org.broadinstitute.hellbender.tools.exome.gcbias;
+
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.DefaultRealMatrixChangingVisitor;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.apache.commons.math3.stat.descriptive.rank.Median;
+import org.broadinstitute.hellbender.tools.exome.ReadCountCollection;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * Stores a list of coverages for each GC content bin
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+class GCCorrector {
+    // GC bins are 0%, 1% . . . 100%
+    private final static int NUMBER_OF_GC_BINS = 101;
+
+    // scale (in units of GCcontent from 0 to 1) over which gc bias correlation decays
+    // i.e. the bias at GC content = 0.3 and at 0.2 are correlated ~exp(-0.1/correlationLength)
+    // this is effectively a smoothness parameter used for regularizing the GC bias estimate for
+    // GC content bins that have few targets)
+    private static final double correlationLength = 0.02;
+    private static final double correlationDecayRatePerBin = 1.0 / (correlationLength * NUMBER_OF_GC_BINS);
+
+    // multiply by these to get a GC correction as a function of GC
+    private final double[] gcCorrectionFactors;
+
+    public GCCorrector(final double[] gcValues, final RealVector coverage) {
+        Utils.nonNull(gcValues);
+        Utils.nonNull(coverage);
+        Utils.validateArg(gcValues.length > 0, "must have at lest one datum");
+        Utils.validateArg(gcValues.length == coverage.getDimension(), "must have one gc value per coverage.");
+
+        final List<List<Double>> coveragesByGC = new ArrayList<>(NUMBER_OF_GC_BINS);
+        IntStream.range(0, NUMBER_OF_GC_BINS).forEach(n -> coveragesByGC.add(new ArrayList<>()));
+        IntStream.range(0, gcValues.length).forEach(n -> coveragesByGC.get(gcContentToBinIndex(gcValues[n])).add(coverage.getEntry(n)));
+        gcCorrectionFactors = calculateCorrectionFactors(coveragesByGC);
+    }
+
+    // weighted average of medians
+    private double[] calculateCorrectionFactors(final List<List<Double>> coveragesByGC) {
```

Comment:
I realize that this is a private method, but a brief description of what is being calculated here (or a reference to something in ``docs/``) would be helpful.